### PR TITLE
support integration test in CI

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

Add integration test support in CI

### What is changed and how it works?

- run integration test in CI
- collect coverage combining integration test and unit test and upload to coveralls.io

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -63,6 +64,11 @@ check_third_party_binary:
 	@which bin/sync_diff_inspector
 	@which bin/go-ycsb
 
+integration_test_build:
+	$(GOTEST) -c -cover -covemode=atomic \
```
Comment (suzaku) [inline Makefile]: Why is `$(GOTEST)` called integration test here?

Comment (amyangfei) [reply Makefile]: Here we use `go test` tool to compile a `cdc binary`, which compiles the package in `main_test.go` and links to the main test program in `main_test.go`. In this way we can run a cdc daemon in the `go test` way.

Comment (suzaku) [reply Makefile]: Maybe add an example of how to use it in the comment?

Comment (amyangfei) [reply Makefile]: already have a reference link in `main_test.go`

**Pull request merged**